lib/pmdtester/parsers/projects_parser.rb

```diff
@@ -0,0 +1,24 @@
+require 'rexml/document'
+require_relative '../project'
+include REXML
+
+module PmdTester
+
+  class ProjectsParser
+    def initialize(list_file)
+      @list_file = list_file
+    end
+
+    def parse
+      list_xml_doc = Document.new(File.new(@list_file))
+      #TODO validate the XML file against an XSD schema
```

Comment:
Uploading the schema file should not be needed. Usually you can refer the the xsd file the same way you refer to the xml file.
I didn't find documentation on REXML, but if you search for "ruby validate xml against xsd" you'll find some alternative libraries, e.g. https://stackoverflow.com/questions/1396759/xsd-schema-validation-in-ruby suggests to use https://rubygems.org/gems/nokogiri/ . nokogiri seems to be still under active development, while rexml is not (https://rubygems.org/gems/rubysl-rexml).

We can upload the schema file later on, as it makes it easier for external xml editors. But it is not need while working on this project - this project will always be the primary and definitive source of the schema.
That's similar to the other schema we have in PMD - while they are available from the web, the primary source is the source code (e.g. https://github.com/pmd/pmd/blob/master/pmd-core/src/main/resources/ruleset_2_0_0.xsd)